src/hooks/usePoolTickData.ts

```diff
@@ -0,0 +1,109 @@
+import { Currency } from '@uniswap/sdk-core'
+import { FeeAmount, nearestUsableTick, TickMath, tickToPrice, TICK_SPACINGS } from '@uniswap/v3-sdk'
+import keyBy from 'lodash.keyby'
+import JSBI from 'jsbi'
+import { PoolState, usePool } from './usePools'
+import { useAllV3Ticks } from './useAllV3Ticks'
+import { useMemo } from 'react'
+import { TickProcessed } from 'constants/ticks'
+import computeSurroundingTicks from 'utils/computeSurroundingTicks'
+
+const PRICE_FIXED_DIGITS = 8
+
+export function usePoolTickData(
+  currencyA: Currency | undefined,
+  currencyB: Currency | undefined,
+  feeAmount: FeeAmount | undefined
+): {
+  loading: boolean
+  error: boolean
+  valid: boolean
+  activeTick: number | undefined
+  tickData: TickProcessed[]
+} {
+  const pool = usePool(currencyA, currencyB, feeAmount)
+
+  const tickSpacing = feeAmount && TICK_SPACINGS[feeAmount]
+
+  // Find nearest valid tick for pool in case tick is not initialized.
+  const activeTick =
+    pool[1]?.tickCurrent && tickSpacing ? nearestUsableTick(pool[1]?.tickCurrent, tickSpacing) : undefined
+
+  const { loading, error, valid, tickData } = useAllV3Ticks(currencyA?.wrapped, currencyB?.wrapped, feeAmount)
+
+  const token0 = currencyA?.wrapped
+  const token1 = currencyB?.wrapped
+
+  return useMemo(() => {
+    if (
+      !token0 ||
+      !token1 ||
+      loading ||
+      error ||
+      !valid ||
+      pool[0] !== PoolState.EXISTS ||
+      activeTick === undefined ||
+      !tickSpacing
+    ) {
+      return {
+        loading: loading || pool[0] === PoolState.LOADING,
+        error: error || pool[0] === PoolState.INVALID,
+        valid: false,
+        activeTick,
+        tickData: [],
+      }
+    }
+
+    const tickToInitializedTick = keyBy(tickData, 'tick')
+
+    // ensure active tick is within supported range
```

Comment:
hmm this should never happen, right?

Reply:
Theoretically  not.. That was taken from @uniswap/uniswap-v3-info. Should I remove it?

Reply:
yeah, let's try :p